### PR TITLE
Fix/pythreejs rotation

### LINKED ISSUE #26: pythreejs coordinate system is wrong
The coordinate system in pythreejs needs to be rotated by 180 degrees around the y axis to match other backends by default.

Sorry for the atrocious lighting - I made it look somewhat reasonable for povray (a little blown out) but it looks pretty bad in all the other backends. Also it looks like both `matplotlib` and `pythreejs` chose a different coordinate system for the directional light than `fresnel` and `povray`.

# pythreejs
image

# fresnel
image

# povray
image

# matplotlib
image

# Generating script:

```python
import hoomd
from hoomd import md
import plato.draw.pythreejs as draw  # or whatever backend
import numpy as np
import matplotlib.cm
import freud
from sklearn.preprocessing import minmax_scale
hoomd.context.initialize('')

# Create a 10x10x10 simple cubic lattice of particles with type name A
system = hoomd.init.create_lattice(unitcell=hoomd.lattice.sc(a=2.0, type_name='A'), n=10)

# Specify Lennard-Jones interactions between particle pairs
nl = md.nlist.cell()
lj = md.pair.lj(r_cut=3.0, nlist=nl)
lj.pair_coeff.set('A', 'A', epsilon=1.0, sigma=1.0)

# Integrate at constant temperature
md.integrate.mode_standard(dt=0.005)
integrator = hoomd.md.integrate.npt(group=hoomd.group.all(), kT=1.2, tau=1, P=5, tauP=1)
integrator.randomize_velocities(seed=42)

# Run for 10,000 time steps
hoomd.run(100e3)

snap = system.take_snapshot()

positions = snap.particles.position
ld = freud.density.LocalDensity(3.0, 1.0, 1.0)
box = freud.box.Box.from_box(snap.box)
ld.compute(box, positions)
radii = np.ones(len(positions)) * 0.5
colors = matplotlib.cm.viridis(minmax_scale(ld.density))
light = np.array([-0.3, -0.3, -0.3])
light /= np.linalg.norm(light)
scene = draw.Scene(draw.Spheres(positions=snap.particles.position, radii=radii, colors=colors),
                   features={'ambient_light': 1, 'directional_light': light},
                   zoom=2)
scene.show()
```

### PR DESCRIPTION
**Pull request by bdice**
This PR aims to resolve #26. Currently the scenes are drawn with the correct rotation, but querying `scene.rotation` gives a wrong answer. This will need to be resolved before merging. I'm quite confused by this issue and would appreciate help.

### CODE REVIEW

File: plato/draw/pythreejs/Scene.py [hunk 1]
```diff
@@ -69,17 +49,24 @@ def zoom(self, value):
     @property
     def rotation(self):
         camera = self._backend_objects['camera']
-        return self._get_camera_quat(camera)
+        norm_out = -np.array(camera.position)
+        norm_out /= np.linalg.norm(norm_out)
+        norm_up = np.array(camera.up)
```
Comment (klarh) [inline plato/draw/pythreejs/Scene.py]: I don't remember all the details of which attributes were actually getting transmitted properly and used by pythreejs with respect to the camera, but the old version of this method only uses `camera.position`, not anything else from the camera. I'd say that's probably why the method isn't working as expected now; did you try with the old version?

**Comment by bdice** [comment]
On 2b0cf5f, I tried to test setting the scene translation to `[3, 5, -50]` for one of the axis scenes but it doesn't appear to do anything unless `scene._update_camera()` is called. However, calling `scene._update_camera()` also changes the camera position and changes the output of the getter for `scene.rotation`.

(Copied from Slack): When `_update_camera` is called, it does the following:
1. Compute the scene size and figure out a distance `dz` for the camera.
2. Rotates `[0, 0, 1 + dz/2]` by `rowan.conjugate(self.rotation)` to compute `translation`
3. Shifts translation by the scene translation: `translation -= self.translation`
4. Moves the camera to this new position, sets rendering bounds and up vector

BUT then the getter `self.rotation` relies on this new camera position to compute its rotation quaternion. *This is wrong because the camera position has been translated by `self.translation` already.*

Subsequent calls to `_update_camera` then get the wrong value for `self.rotation` because they rely on a post-translated camera position.

Possible solution: Use `camera.position - self.translation` in the getter for `self.rotation`. However, I don't know how to account for changes introduced by the user panning the camera -- those don't affect `self.translation` to my knowledge.

**Comment by klarh** [comment]
It looks like the `target` attribute of the controls object contains the translation state of the scene when you control-grab to drag around the view (which is applied before the camera rotation). I think we should be able to use that to properly determine both the orientation and translation of the scene.

File: plato/draw/pythreejs/Scene.py [hunk 2]
```diff
@@ -66,20 +50,50 @@ def zoom(self):
...
-        rotation = math.quatquat(new_quat, math.quatconj(old_quat))
-
+        conj = rowan.conjugate(value)
         camera = self._backend_objects['camera']
-        new_position = math.quatrot(rotation, camera.position)
-        camera.position = new_position.tolist()
+
+        # rotating a scene by q is equivalent to rotating the camera's
+        # position and up vector by q*
+        conj = rowan.conjugate(value)
```
Comment (bdice) [inline plato/draw/pythreejs/Scene.py]: This line is duplicated above.

**Comment by bdice** [comment]
@klarh I think this is ready for a final review and merge.
